docs/proposals/proposal-003-report.md

```diff
@@ -0,0 +1,254 @@
+-->
+
+The main risks are that the metrics captured and recorded aren't useful or don't show much.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate?
+Think broadly.  For example, consider how this will impact or be impacted
+by scaling to support more CNCF Projects.
+
+How will this affect the benchmark tests, CNCF Project Maintainers, pipeline maintainers, etc?
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include manifests or workflow examples
+(though not always required) or even code snippets. If there's any ambiguity
+about HOW your proposal will be implemented, this is the place to discuss them.
+-->
+
+### Setup
+
+1. Start Kubernetes
+2. [Install and start Prometheus](https://sustainable-computing.io/installation/kepler/#deploy-the-prometheus-operator) 2. `cd kube-prometheus` 3. `kubectl apply --server-side -f manifests/setup` 4. `kubectl apply -f manifests/` 5. Wait… 6. `kubectl -n monitoring port-forward svc/grafana 3000` 7. Open dashboard _localhost:3000_
+3. Install metrics server
+   1. `kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml`
+   2. Patch
+
+```shell
+kubectl patch -n kube-system deployment metrics-server --type=json \
+-p '[{"op":"add","path":"/spec/template/spec/containers/0/args/-","value":"--kubelet-insecure-tls"}]'
+```
+
+4. Install and start Kepler
+   1. Open a new terminal
+   2. `git clone --depth 1 git@github.com:sustainable-computing-io/kepler.git`
+5. `cd kepler`
+6. `make build-manifest OPTS="PROMETHEUS_DEPLOY"`
+7. `kubectl apply -f _output/generated-manifest/deployment.yaml`
+8. Add [dashboard](https://raw.githubusercontent.com/sustainable-computing-io/kepler/main/grafana-dashboards/Kepler-Exporter.json) to Grafana.
+9. Install and start Falco
+   1. Open a new terminal
+   2. [Install Helm](https://helm.sh/docs/intro/install/)
+   3. `helm repo add falcosecurity https://falcosecurity.github.io/charts`
+   4. `helm repo update`
+   5. `helm install falco falcosecurity/falco --namespace falco --create-namespace --set driver.kind=modern-bpf  --set falco.grpc.enabled=true --set falco.grpc_output.enabled=true`
+   6. `helm install falco-exporter falcosecurity/falco-exporter`
+10. Run Falco tests
+    1. https://github.com/falcosecurity/cncf-green-review-testing/tree/main/benchmark-tests
+11. May need to remove `nodeSelector`
+12. Write out metrics to JSON
+13. Thinking about https://github.com/prometheus/prom2json
+14. These metrics:
+
+    ```
+    rate(container_cpu_usage_seconds_total[5m])
+    container_memory_rss
+    container_memory_working_set_bytes
+    kepler_container_joules_total
+    ```
+
+## Drawbacks (Optional)
+
+JSON storage isn't so viable in the long term, this is an MVP solution and in the long term, will store metrics in a Prometheus-compatible storage solution.
```

Comment:
@AntonioDiTuri @ChrisChinchilla I thought some more on this after our discussion in the meeting.

For the first iteration I think exporting the metrics to JSON and committing to a git branch is good. Due to team size we want to move in smaller iterations and this achieves our milestone of having the pipeline complete end to end.

The added benefit is we have a copy of the results outside of the cluster.

For the second iteration I think an important consideration is how can we present the results in an appealing and usable way. Since we have Grafana and Prometheus in the cluster that is an option. A drawback is it needs
a cluster with at least 1 node running 24/7.

Reply:
I am fine with either and bow to superior knowledge of the two of you, just let me know what approach you prefer and I'll continue.

Reply:
True @rossf7, making something simple can help us move forward and in a next iteration we can improve it to allow for visualisation and working with the data...